at_client/lib/src/client/BatchVerbBuilder.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:at_client/at_client.dart';
+import 'package:at_client/src/service/encryption_service.dart';
+import 'package:at_commons/at_builders.dart';
+import 'package:at_commons/at_commons.dart';
+import 'package:at_commons/src/keystore/at_key.dart';
+import 'package:at_utils/at_utils.dart';
+
+class BatchVerbBuilder {
+  List<BatchRequest> _batchRequests = [];
+  var _batchId;
+  var _currentAtSign;
+  EncryptionService _encryptionService;
+
+  BatchVerbBuilder(int batchId, String atSign) {
+    this._currentAtSign = atSign;
+    this._batchId = batchId;
+  }
+
+  Future<void> delete(AtKey atKey) async {
+    var isPublic = atKey.metadata != null ? atKey.metadata.isPublic : false;
+    var isNamespaceAware =
+        atKey.metadata != null ? atKey.metadata.namespaceAware : true;
+    AtClientImpl atClient = await AtClientImpl.getClient(_currentAtSign);
+    var namespace = atClient.preference.namespace;
+    var keyWithNamespace;
+    if (isNamespaceAware) {
+      keyWithNamespace = _getKeyWithNamespace(atKey.key, namespace);
+    } else {
+      keyWithNamespace = atKey.key;
+    }
+    var builder = DeleteVerbBuilder()
+      ..isPublic = isPublic
+      ..sharedWith = atKey.sharedWith
+      ..atKey = keyWithNamespace
+      ..sharedBy = _currentAtSign;
+    _addToList(builder);
+  }
+
+  String _getKeyWithNamespace(String key, String namespace) {
+    var keyWithNamespace = key;
+    if (namespace != null && namespace.isNotEmpty) {
+      keyWithNamespace += '.${namespace}';
+    }
+    return keyWithNamespace;
+  }
+
+  void get(AtKey atKey) async {
+    var builder;
+    var keyWithNamespace;
+    var isPublic = atKey.metadata != null ? atKey.metadata.isPublic : false;
+    var namespaceAware =
+        atKey.metadata != null ? atKey.metadata.namespaceAware : true;
+    var isCached = atKey.metadata != null ? atKey.metadata.isCached : false;
+    AtClientImpl atClient = await AtClientImpl.getClient(_currentAtSign);
+    var namespace = atClient.preference.namespace;
+    var key = atKey.key;
+    var sharedBy = atKey.sharedBy;
+    var sharedWith = atKey.sharedWith;
+    if (namespaceAware) {
+      keyWithNamespace = _getKeyWithNamespace(key, namespace);
+    } else {
+      keyWithNamespace = key;
+    }
+    if (sharedBy != null && isCached) {
```

Comment:
below logic introduces code duplication..we have to think of some other approach